### PR TITLE
chore: new db migration tables

### PR DESCRIPTION
**Pull request by igkostyuk**
migrations with new tables
https://dbdiagram.io/d/610d1c2d2ecb310fc3c078b7

feel free to add any change for discussion

### CODE REVIEW

**Review by KekemonBS — APPROVED** [review]

File: migrations/0002_06.08.2021-migrations.down.sql [hunk 1]
```diff
@@ -0,0 +1,16 @@
+-- +migrate Down
+DROP TABLE IF EXISTS users cascade;
```
Comment (glbter) [inline migrations/0002_06.08.2021-migrations.down.sql:2]: when deleting all db, cascade will only make it longer, the result will be same

File: migrations/0002_06.08.2021-migrations.up.sql [hunk 2]
```diff
@@ -0,0 +1,61 @@
+CREATE TABLE IF NOT EXISTS roles (
+   id serial PRIMARY KEY,
+   name varchar NOT NULL,
+   description varchar NOT NULL
+);
+
+CREATE TABLE IF NOT EXISTS permisions (
+   id serial PRIMARY KEY,
+   name varchar NOT NULL,
+   description varchar NOT NULL,
```
Comment (glbter) [inline migrations/0002_06.08.2021-migrations.up.sql]: do we need descriptions? I can't think up an examle with non-descriptive role, permission, schedule

**Review by glbter — CHANGES_REQUESTED** [review]

File: migrations/0002_06.08.2021-migrations.up.sql [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+   wed tsrange,
+   thu tsrange,
+   fri tsrange,
+   sat tsrange
+);
+
+CREATE TABLE IF NOT EXISTS doctors (
+   id uuid PRIMARY KEY DEFAULT uuid_generate_v4 (),
+   user_id uuid REFERENCES users (id),
+   full_name varchar(50),
```
Comment (glbter) [inline migrations/0002_06.08.2021-migrations.up.sql]: if full_name is first_name + last_name + patronimic, it is not quite first normal form

**Review by glbter — CHANGES_REQUESTED** [review]

**Comment by glbter** [comment]
@igkostyuk `varchar` without n and `text` have **no performance difference**, 
char is static length type
[resource](https://www.postgresqltutorial.com/postgresql-char-varchar-text/)

File: migrations/0002_06.08.2021-migrations.up.sql [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+   role_id int REFERENCES roles (id)
+);
+
+CREATE TABLE IF NOT EXISTS cards (
+   id uuid PRIMARY KEY
+);
+
+CREATE TABLE IF NOT EXISTS patients (
+   id uuid PRIMARY KEY,
+   full_name varchar(50),
```
Comment (VovaD199) [inline migrations/0002_06.08.2021-migrations.up.sql]: if users have a name, why is there "full name" and not "second name"

**Review by VovaD199 — APPROVED** [review]

File: migrations/0002_06.08.2021-migrations.down.sql (hunk 1 above)
Comment (igkostyuk) [reply migrations/0002_06.08.2021-migrations.down.sql:2]: don't think we will  feel any performance issues here

File: migrations/0002_06.08.2021-migrations.up.sql (hunk 2 above)
Comment (igkostyuk) [reply migrations/0002_06.08.2021-migrations.up.sql]: what about the Operator role ?

File: migrations/0002_06.08.2021-migrations.up.sql (hunk 3 above)
Comment (igkostyuk) [reply migrations/0002_06.08.2021-migrations.up.sql]: completely agree

File: migrations/0002_06.08.2021-migrations.up.sql (hunk 4 above)
Comment (igkostyuk) [reply migrations/0002_06.08.2021-migrations.up.sql]: the user name is not a first name or last name,  I will change it to first and last names

**Comment by glbter** [comment]
if we are not deleting appointments we should make id a timestamp, so it would be descriptive and faster to find. With doctors it is harder to use key pagination. As you were right - the sql `offset` only skips records, wich will be to long in a big table @igkostyuk

**Comment by igkostyuk** [comment]
> @igkostyuk `varchar` without n and `text` have **no performance difference**,
> char is static length type
> [resource](https://www.postgresqltutorial.com/postgresql-char-varchar-text/)

@glbter so what should we do?

**Comment by glbter** [comment]
> @glbter so what should we do?

I think, try to find an upper limit for our strings such a names, etc. Like name is varchar(16) or something

**Comment by igkostyuk** [comment]
> > @glbter so what should we do?
> 
> I think, try to find an upper limit for our strings such a names, etc

@glbter any suggestions?

**Comment by igkostyuk** [comment]
> > @glbter so what should we do?
> 
> I think, try to find an upper limit for our strings such a names, etc. Like name is varchar(16) or something

@glbter so how it will be i'm confused?

**Comment by glbter** [comment]
> 
> 
> > > @glbter so what should we do?
> > 
> > 
> > I think, try to find an upper limit for our strings such a names, etc
> 
> @glbter any suggestions?
@igkostyuk 

- password hash should be static, as the hash is, I cant remember how long it is
- names and last names maybe abot 15-25, as you wish
- roles names we will find out, descriptions aren't so big
- only in appointments, it should be 100-150
- doctors specialities we will find out, maybe 25 is more than enough

**Review by glbter — APPROVED** [review]

**Review by ghost — APPROVED** [review]
all ok

**Review by metallrooch — APPROVED** [review]

**Pull request merged**